recipes-kernel/linux/files/patches-5.10/0121-serial-core-Initialize-rs485-RTS-polarity-already-on.patch

```diff
@@ -0,0 +1,144 @@
+From: Lukas Wunner <lukas@wunner.de>
+Date: Sun, 23 Jan 2022 05:21:14 +0100
+Subject: [PATCH 121/121] serial: core: Initialize rs485 RTS polarity already
+ on probe
+
+RTS polarity of rs485-enabled ports is currently initialized on uart
+open via:
+
+tty_port_open()
+  tty_port_block_til_ready()
+    tty_port_raise_dtr_rts()  # if (C_BAUD(tty))
+      uart_dtr_rts()
+        uart_port_dtr_rts()
+
+There's at least three problems here:
+
+First, if no baud rate is set, RTS polarity is not initialized.
+That's the right thing to do for rs232, but not for rs485, which
+requires that RTS is deasserted unconditionally.
+
+Second, if the DeviceTree property "linux,rs485-enabled-at-boot-time" is
+present, RTS should be deasserted as early as possible, i.e. on probe.
+Otherwise it may remain asserted until first open.
+
+Third, even though RTS is deasserted on open and close, it may
+subsequently be asserted by uart_throttle(), uart_unthrottle() or
+uart_set_termios() because those functions aren't rs485-aware.
+(Only uart_tiocmset() is.)
+
+To address these issues, move RTS initialization from uart_port_dtr_rts()
+to uart_configure_port().  Prevent subsequent modification of RTS
+polarity by moving the existing rs485 check from uart_tiocmget() to
+uart_update_mctrl().
+
+That way, RTS is initialized on probe and then remains unmodified unless
+the uart transmits data.  If rs485 is enabled at runtime (instead of at
+boot) through a TIOCSRS485 ioctl(), RTS is initialized by the uart
+driver's ->rs485_config() callback and then likewise remains unmodified.
+
+The PL011 driver initializes RTS on uart open and prevents subsequent
+modification in its ->set_mctrl() callback.  That code is obsoleted by
+the present commit, so drop it.
+
+Cc: Jan Kiszka <jan.kiszka@siemens.com>
+Cc: Su Bao Cheng <baocheng.su@siemens.com>
+Signed-off-by: Lukas Wunner <lukas@wunner.de>
+Link: https://lore.kernel.org/r/2d2acaf3a69e89b7bf687c912022b11fd29dfa1e.1642909284.git.lukas@wunner.de
+Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
+---
+ drivers/tty/serial/amba-pl011.c  | 10 +++++++---
+ drivers/tty/serial/serial_core.c | 34 +++++++++++---------------------
+ 2 files changed, 19 insertions(+), 25 deletions(-)
+
+diff --git a/drivers/tty/serial/amba-pl011.c b/drivers/tty/serial/amba-pl011.c
+index 61183e7ff009..0162e9140035 100644
+--- a/drivers/tty/serial/amba-pl011.c
++++ b/drivers/tty/serial/amba-pl011.c
+@@ -1763,9 +1763,13 @@ static int pl011_startup(struct uart_port *port)
+ 
+ 	spin_lock_irq(&uap->port.lock);
+ 
+-	/* restore RTS and DTR */
+-	cr = uap->old_cr & (UART011_CR_RTS | UART011_CR_DTR);
+-	cr |= UART01x_CR_UARTEN | UART011_CR_RXE | UART011_CR_TXE;
++	cr = pl011_read(uap, REG_CR);
++	cr &= UART011_CR_RTS | UART011_CR_DTR;
++	cr |= UART01x_CR_UARTEN | UART011_CR_RXE;
++
++	if (!(port->rs485.flags & SER_RS485_ENABLED))
++		cr |= UART011_CR_TXE;
++
+ 	pl011_write(cr, uap, REG_CR);
+ 
+ 	spin_unlock_irq(&uap->port.lock);
```

Comment:
This part is not in the official backport for 5.10: https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?id=a1508d164e58dc0857a1103a014d78f321d7481f

Reply:
Or here: https://www.spinics.net/lists/stable/msg571707.html

Reply:
thanks . I would repick the commit